Sources/ConfidenceProvider/Cache/Models/CacheData.swift

```diff
@@ -33,7 +33,27 @@ struct CacheData: Codable {
         return resolveTokenIndex != nil
     }
 
-    mutating func add(resolveToken: String, flagName: String, applyTime: Date) {
+    mutating func setEventStatus(resolveToken: String, name: String, status: ApplyEventStatus = .sent) {
+        let flagEventIndexes = flagEventIndex(resolveToken: resolveToken, name: name)
+        guard let resolveIndex = flagEventIndexes.resolveEventIndex,
+              let flagIndex = flagEventIndexes.flagEventIndex else {
+            return
+        }
+
+        resolveEvents[resolveIndex].events[flagIndex].status = status
+    }
+
+    mutating func setEventStatus(resolveToken: String, status: ApplyEventStatus = .sent) {
+        guard let resolveIndex = resolveEventIndex(resolveToken: resolveToken) else {
+            return
+        }
+
+        for i in 0...resolveEvents[resolveIndex].events.count-1 {
```

Comment:
nit: `for i in 0..<resolveEvents[resolveIndex].events.count`

Reply:
Good point. Fixed this ✅